# Titanic Data Analysis

A beginner exploratory data analysis project using the Kaggle Titanic training dataset.

In [33]:
import pandas as pd

df = pd.read_csv("train.csv")
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [34]:
df.shape

(891, 12)

In [35]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [36]:
df.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


In [37]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

## Analysis Questions

1. How many rows and columns are in the dataset?

In [38]:
df.shape

(891, 12)

rows = 891
columns = 12

2. What are all column names?

In [39]:
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

3. Which columns have missing values, and how many?

In [40]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

Age(177), Cabin(687) and Embarked(2) Columns have missing values.

4. What is the average passenger age?

In [41]:
df.Age.mean().round(2)

np.float64(29.7)

The average Age is 29.7 years

5. How many passengers survived?

In [42]:
df.Survived.sum()

np.int64(342)

Total 342 passengers survived.

6. What percentage of passengers survived?

(df.Survived.sum()/len(df) * 100).round(2)

38.38 % passengers survived

7. What is the survival rate for female versus male passengers?

In [43]:
df.groupby("Sex").Survived.mean() * 100

Sex
female    74.203822
male      18.890815
Name: Survived, dtype: float64

8. What is the survival rate for each passenger class (Pclass)?

In [44]:
df.groupby("Pclass").Survived.mean() * 100

Pclass
1    62.962963
2    47.282609
3    24.236253
Name: Survived, dtype: float64

9. What is the average fare for each passenger class?

In [45]:
df.groupby("Pclass").Fare.mean()

Pclass
1    84.154687
2    20.662183
3    13.675550
Name: Fare, dtype: float64

10. Who was the oldest passenger and who was the youngest passenger?

In [46]:
df.Age.agg(['max', 'min'])

max    80.00
min     0.42
Name: Age, dtype: float64

11. Does survival change by Embarked location?

In [48]:
df.groupby("Embarked").Survived.mean() * 100

Embarked
C    55.357143
Q    38.961039
S    33.695652
Name: Survived, dtype: float64

Embarked = 'C' has the highest survival rate.

12. Compare survival for women in first, second, and third class.

In [50]:
df.groupby(["Sex", "Pclass"]).Survived.mean()["female"]

Pclass
1    0.968085
2    0.921053
3    0.500000
Name: Survived, dtype: float64

## Key Findings

- Finding 1: More Females survived more than the males
- Finding 2: Fare of the 1-Class is much high as compared to 2 and 3 classes
- Finding 3: Ages from 80 to 0.42 were onboarded

## What I Learned

- How to load a CSV dataset into a pandas DataFrame.
- How to identify missing values using `isnull().sum()'
- How to use `groupby()` to compare survival rates by gender and passenger class.